confgenerator/logging_receivers.go

```diff
@@ -738,6 +738,150 @@ func windowsEventLogV1Processors(ctx context.Context) ([]otel.Component, error)
 	return p.Processors(ctx)
 }
 
+// LoggingProcessorWindowsEventLogV2 contains the otel logging processors for ReceiverVersion=2.
+type LoggingProcessorWindowsEventLogV2 struct {
+	ConfigComponent `yaml:",inline"`
+}
+
+func (r LoggingProcessorWindowsEventLogV2) Type() string {
+	return "windows_event_log_v2"
+}
+
+func (p LoggingProcessorWindowsEventLogV2) Components(ctx context.Context, tag, uid string) []fluentbit.Component {
+	// This processor is only intended for otel logging since fluent-bit "winevtlog" receiver generates a specific log structure.
+	return noFluentBitImplementation(ctx, tag, uid)
+}
+
+func (p LoggingProcessorWindowsEventLogV2) Processors(ctx context.Context) ([]otel.Component, error) {
+	return windowsEventLogV2Processors(ctx)
+}
+
+func windowsEventLogV2Processors(ctx context.Context) ([]otel.Component, error) {
+	// The winevtlog input in fluent-bit has a completely different structure.
+	// We need to convert the OTel format into the fluent-bit format.
+	var empty string
+	p := &LoggingProcessorModifyFields{
+		EmptyBody: true,
+		Fields: map[string]*ModifyField{
+			"jsonPayload.Channel":       {CopyFrom: "jsonPayload.channel"},
+			"jsonPayload.Computer":      {CopyFrom: "jsonPayload.computer"},
+			"jsonPayload.EventID":       {CopyFrom: "jsonPayload.event_id.id"},
+			"jsonPayload.EventRecordID": {CopyFrom: "jsonPayload.record_id"},
+			"jsonPayload.Keywords":      {CopyFrom: "jsonPayload.keywords"},
+			"jsonPayload.Level": {
+				CopyFrom: "jsonPayload.level",
+				MapValues: map[string]string{
+					"Critical":    "1",
+					"Error":       "2",
+					"Warning":     "3",
+					"Information": "4",
+				},
+				Type:               "integer",
+				MapValuesExclusive: true,
+			},
+			"jsonPayload.Message":      {CopyFrom: "jsonPayload.message"},
+			"jsonPayload.Opcode":       {CopyFrom: "jsonPayload.opcode", Type: "integer"},
+			"jsonPayload.ProcessID":    {CopyFrom: "jsonPayload.execution.process_id", Type: "integer"},
+			"jsonPayload.ProviderGuid": {CopyFrom: "jsonPayload.provider.guid"},
+			"jsonPayload.ProviderName": {CopyFrom: "jsonPayload.provider.name"},
+			"jsonPayload.Qualifiers":   {CopyFrom: "jsonPayload.event_id.qualifiers"},
+			"jsonPayload.StringInserts": {
+				CopyFrom: "jsonPayload.event_data",
+				CustomConvertFunc: func(v ottl.LValue) ottl.Statements {
+					eventData := ottl.LValue{v.String(), "data"}
+					eventBinary := ottl.LValue{v.String(), "binary"}
+					cacheEventData := ottl.LValue{"cache", "__event_data"}
+					return ottl.NewStatements(
+						cacheEventData.SetIf(ottl.ToValues(eventData), eventData.IsPresent()),
+						cacheEventData.AppendValuesIf(eventBinary, ottl.And(cacheEventData.IsPresent(), eventBinary.IsPresent())),
```

Comment:
Can you test what happens if the record _only_ has `Binary` and not `Data`? Will `AppendValuesIf` fail if the preceding `SetIf` doesn't end up setting anything on `cacheEventData`?

Reply:
Nevermind, according to the docs it'll create the field if it doesn't exist when appending, and I got curious and tried it myself and it seems to indeed work that way.